sgt-back-end/index.js

```diff
@@ -0,0 +1,221 @@
+const express = require('express');
+const app = express();
+// const fs = require('fs');
+const pg = require('pg');
+
+// pnly create one pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('The server is now listening');
+});
+
+// express.json parses incoming requests with JSON payloads
+const jsonParse = express.json();
+app.use('/api/grades', jsonParse);
+
+app.get('/api/grades', (req, res) => {
+  // first we need to query the database for the data we want
+  // use template literal format so it is easier to indent
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+  `;
+  db.query(sql)
+  // the query succeeded even if nothing was found
+  // the Result object will include an array of rows
+    .then(result => {
+      const grades = result.rows;
+      // this will send the response of the array of objects
+      // back to the client that requested the data
+      res.json(grades);
+    })
+    .catch(err => {
+      // the query failed for some reason
+      // possibly due to syntax error in the sql statement
+      // print the error to the STDERR (terminal) for debugging purposes
+      console.error(err);
+      // respond to the client with a generic 500 error message
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const newGrade = req.body;
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = (req.body.score);
+
+  if (newGrade === {}) {
+    res.status(400).json({
+      error: 'need to include: name, course, and score'
+    });
+    return;
```

Comment:
This condition isn't actually catching anything because no two objects ever equal one another in JavaScript, so you can remove it from anywhere you have this comparison in your code.

Reply:
man I almost didn't do that too. I did the same thing last time.